### ISSUE TITLE
User Defined VPN

### ISSUE THREAD

**Issue by yomatzi** [Zomboided/service.vpn.manager#27]
Hi Zomboided,

I'm trying for ours now, but can't get it working.

The VPN provider is Incloak, I have a .ovpn file, the user .crt and .key file and a ca.crt file.

During configuration the addon asks me for the .ovpn, user.crt and user.key file, but not for the ca.crt file. The validation ends without connecting, in the log I get the error message "Options error: --ca fails with 'ca.crt': No such file or directory". 

I don't understand the problem at all, your help is really appreciated, thanks.

**Comment by Zomboided** [comment]
I'm assuming you've followed the instructions here
https://github.com/Zomboided/service.vpn.manager/blob/master/README.md

The fourth paragraph is the thing that will help you.  On the ca line change it to ca #PATH/ca.crt, then in the settings GUI, go to the utilities tab and reset the ovpn files.  Then try connecting again and it should work.

To debug you can go to /storage/.kodi/addons/service.vpn.manager and see how the ovpn file is resolved after it's copied across for use.  The error message is basically saying it can't find the file so you need to make sure the path it's using is correct.

**Comment by yomatzi** [comment]
Thank you, but I was not clear enough, sorry. I already changed the three 
lines in the ovpn file (ca.crt, user.key, user.crt) and did the reset. I 
have read all your instructions on github.

I couldn't debug and check the resolved ovpn file, because there is no 
folder /storage/.kodi/addons/ on my OS (which is strange and I don't know 
why). I have Libreelec 7.1.1 with Kodi Jarvis for Odroid C2.

Gesendet mit AquaMail für Android
http://www.aqua-mail.com


--- Original Nachricht ---
Von: Zomboided <notifications@github.com>
Datum: 31. Dezember 2016 12:23:45
Betreff: Re: [Zomboided/service.vpn.manager] User Defined VPN (#27)
An: Zomboided/service.vpn.manager <service.vpn.manager@noreply.github.com>
CC: yomatzi <yomatzi@gmx.de>, Author <author@noreply.github.com>

**Comment by Zomboided** [comment]
Ok, I use a different platform. Up to /addons, the path will be different (it's then resolved using a call internally to let it work on multiple platforms). You need to find this directory, then navigate to the service.vpn.manager/UserDefined directory and see what's there/what's in the ovpn file

**Comment by yomatzi** [comment]
Thank you, it's working now, but the way was a little different than you described it.

I found the addons directory (same path as you wrote, but it didn't show up automatically via samba because it's hidden). I checked the created ovpn file there. The line with the ca.crt file wasn't changed to the other path, so I've done that manually. But resetting the ovpn file in the utilities tab deletes the created ovpn file, so the changes on the file were irrelevant (this is not clear for me in your instructions). As a side note, adding the tag "#PATH" in the original ovpn file didn't work for me.

Thank you again ;) !!!
